### PR TITLE
Calkim duo detections

### PR DESCRIPTION
**Pull request by calkim-panther**
### Background
New detections:
- Admin Create
- MFA restrictions Updated
- new admin api integration
- admin policy updated
- sso saml requirement disabled
- admin user mfa bypass 

### Testing
```Duo.Admin.Create.Admin
	[PASS] Admin Create
		[PASS] [rule] true
		[PASS] [title] Duo: [Bart Simpson] created a new admin account for [Homer Simpson] [homer.simpson@simpsons.com].
		[PASS] [dedup] Duo: [Bart Simpson] created a new admin account for [Homer Simpson] [homer.simpson@simpsons.com].
		[PASS] [alertContext] {"action": "admin_create", "description": "{\"name\": \"Homer Simpson\", \"phone\": null, \"is_temporary_password\": false, \"email\": \"homer.simpson@simpsons.com\", \"hardtoken\": null, \"role\": \"Owner\", \"status\": \"Pending Activation\", \"restricted_by_admin_units\": false, \"administrative_units\": \"\"}", "username": "Bart Simpson", "timestamp": "2023-01-17 16:47:54"}
	[PASS] Other Event
		[PASS] [rule] false

Duo.Admin.MFA.Restrictions.Updated
	[PASS] Admin MFA Update Event
		[PASS] [rule] true
		[PASS] [title] Duo Admin MFA Restrictions Updated by [Homer Simpson]
		[PASS] [dedup] Duo Admin MFA Restrictions Updated by [Homer Simpson]
		[PASS] [alertContext] {"action": "update_admin_factor_restrictions", "description": "{\"allowed_factors\": \"Duo mobile passcodes, Hardware tokens, Duo push, Yubikey aes\"}", "username": "Homer Simpson", "timestamp": "2022-02-21 21:48:06"}
	[PASS] Login Event
		[PASS] [rule] false

Duo.Admin.New.Admin.API.App.Integration
	[PASS] Admin API Integration Created
		[PASS] [rule] true
		[PASS] [title] Duo: [Homer Simpson] created a new Admin API integration to [Admin API]
		[PASS] [dedup] Duo: [Homer Simpson] created a new Admin API integration to [Admin API]
		[PASS] [alertContext] {"action": "integration_create", "description": "{\"greeting\": \"\", \"notes\": \"\", \"offline_auth_enabled\": 0, \"offline_max_days\": 0, \"offline_max_attempts\": 0, \"type\": \"Admin API\", \"raw_type\": \"adminapi\", \"name\": \"Admin API\", \"self_service_allowed\": false, \"username_normalization_policy\": \"None\", \"missing_web_referer_policy\": \"deny\", \"networks_for_api_access\": \"\", \"group_access\": \"\"}", "username": "Homer Simpson", "timestamp": "2021-11-30 17:15:33"}
	[PASS] Non Admin API Integration
		[PASS] [rule] false
	[PASS] Other Event
		[PASS] [rule] false

Duo.Admin.Policy.Updated
	[PASS] Policy Update
		[PASS] [rule] true
		[PASS] [title] Duo: [Homer Simpson] updated [Global Policy].
		[PASS] [dedup] Duo: [Homer Simpson] updated [Global Policy].
		[PASS] [alertContext] {"action": "policy_update", "description": "{\"adaptive_auth_display_unit\": \"days\", \"trusted_mobile_endpoint_policy\": \"no action\", \"adaptive_auth_hours\": 0, \"admin_email\": \"homer.simpson@simpsons.com\", \"allow_factor_u2f\": false, \"device_certificate_policy\": \"no action\", \"allow_factor_phone\": false, \"local_trusted_sessions_display_val\": 0, \"allow_adaptive_auth\": false, \"local_trusted_sessions_display_unit\": \"days\", \"allow_factor_sms\": false}", "username": "Homer Simpson", "timestamp": "2022-02-21 21:48:48"}
	[PASS] Other event
		[PASS] [rule] false


Duo.Admin.SSO.SAML.Requirement.Disabled
	[PASS] Enforcement Disabled
		[PASS] [rule] true
		[PASS] [title] Duo: [Homer Simpson] changed SAML authentication requirements for Administrators to [disabled]
		[PASS] [dedup] Duo: [Homer Simpson] changed SAML authentication requirements for Administrators to [disabled]
		[PASS] [alertContext] {"action": "admin_single_sign_on_update", "description": "{\"enforcement_status\": \"disabled\"}", "username": "Homer Simpson", "timestamp": "2021-10-12 21:29:22"}
	[PASS] Enforcement Optional
		[PASS] [rule] true
		[PASS] [title] Duo: [Homer Simpson] changed SAML authentication requirements for Administrators to [optional]
		[PASS] [dedup] Duo: [Homer Simpson] changed SAML authentication requirements for Administrators to [optional]
		[PASS] [alertContext] {"action": "admin_single_sign_on_update", "description": "{\"enforcement_status\": \"optional\"}", "username": "Homer Simpson", "timestamp": "2021-10-12 21:29:22"}
	[PASS] Enforcement Required
		[PASS] [rule] false
	[PASS] SSO Update
		[PASS] [rule] false

Duo.Admin.User.MFA.Bypass.Enabled
	[PASS] Account Active
		[PASS] [rule] false
	[PASS] Account Disabled
		[PASS] [rule] false
	[PASS] Bypass Enabled
		[PASS] [rule] true
		[PASS] [title] Duo: [Homer Simpson] updated account [bart.simpson@simpsons.com]  to not require two-factor authentication.
		[PASS] [dedup] Duo: [Homer Simpson] updated account [bart.simpson@simpsons.com]  to not require two-factor authentication.
		[PASS] [alertContext] {"action": "user_update", "description": "{\"status\": \"Bypass\"}", "username": "Homer Simpson", "timestamp": "2021-10-05 22:45:33"}
	[PASS] Phones Update
		[PASS] [rule] false
		
Panther CLI Test Summary
	Path: rules/duo_rules
	Passed: 15
	Failed: 0
	Invalid: 0
```

### CODE REVIEW

**Comment by jpl5280** [comment]
@calkim-panther - can you put the output from CLI testing in the PR description? Helps to see the dynamically generated titles, etc

File: rules/duo_rules/duo_admin_user_mfa_bypass_enabled.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        if "status" in description:
+            return description.get("status") == "Bypass"
+    return False
+
+
+def title(event):
+    return (
+        f"Duo: [{event.get('username', '<username_not_found>')}] "
+        f"updated account [{event.get('object', '<object_not_found>')}] "
+        " to not require two-factor authentication."
```
Comment (jpl5280) [inline rules/duo_rules/duo_admin_user_mfa_bypass_enabled.py:19]: extra space

Comment (jpl5280) [reply rules/duo_rules/duo_admin_user_mfa_bypass_enabled.py:19]:
```suggestion
        "to not require two-factor authentication."
```

File: rules/duo_rules/duo_admin_create_admin.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+def rule(event):
+    return event.get("action") == "admin_create"
+
+
+def title(event):
+    event_description = deserialize_administrator_log_event_description(event)
+    return (
+        f"Duo: [{event.get('username', '<username_not_found>')}] "
+        "created a new admin account for "
```
Comment (jpl5280) [inline rules/duo_rules/duo_admin_create_admin.py:15]:
```suggestion
        "created a new admin account: "
```

**Review by jpl5280 — CHANGES_REQUESTED** [review]

**Review by andrea-youwakim — APPROVED** [review]

**Review by jpl5280 — APPROVED** [review]

**Pull request merged**